M3-2581 Add GA event for compact mode switcher

WilkinsKa1 opened the pull request:
## Description

Added a GA event for the spacing switcher:
category: 'Theme Choice',
action: 'Spacing Toggle',
label: 'compact' | 'normal'

## Type of Change
- Non breaking change ('update', 'change')

If the any above types of change apply to this pull request, please ensure to include one of the listed keywords in the pull request title.

martinmckenna commented:
This is more to start a conversation, but do we think it would be more beneficial to send the analytic event of page load or on the clicking of the switch?

Currently, the dark/light theme event gets sent on page load, so I think this should be consistent with that, but we should decide if we want to do both on page load or on button click

Jskobos commented:
> This is more to start a conversation, but do we think it would be more beneficial to send the analytic event of page load or on the clicking of the switch?
> 
> Currently, the dark/light theme event gets sent on page load, so I think this should be consistent with that, but we should decide if we want to do both on page load or on button click

I'm in favor of sending events on toggle, since that's more of an "event", then using some other method to figure out how many people on average are using Dark Theme etc. But I don't have any good ideas on how to do that...might be possible to add the current themeChoice variable as metadata for normal page views.

WilkinsKa1 commented:
I thought the action of clicking the toggle would be best to indicate an active choice on the user's part to customize their theme. I could adjust the current theme event for the dark/light theme toggle to be a click event vs page load?

martinmckenna commented:
@WilkinsKa1 I'm all for tracking both. I'd rather not deprecate what we have but instead add on.

Maybe we could have both? Page load and button click?

My issue with only tracking button clicks is that we're not really gauging what setting people are actively using. In other words, if 20 people click "compact" mode, we're going to get 20 analytic events, but it doesn't really tell us who is using what setting when they browse the app.

I'm of the opinion we track both button clicks and page load settings.

WilkinsKa1 commented:
@martinmckenna I added a click event for the theme toggle and adjusted the page view to combine both theme and spacing values into one rather than having two separate pageview events. Let me know what you think!

martinmckenna commented on src/index.tsx:
```diff
@@ -22,7 +22,7 @@ import { sendEvent } from 'src/utilities/analytics';
 import 'src/utilities/request';
 import isPathOneOf from 'src/utilities/routing/isPathOneOf';
-import { theme } from 'src/utilities/storage';
+import { spacing, theme } from 'src/utilities/storage';
```
This is causing a `shadowed variable` tslint warning on line 71

WilkinsKa1 replied:
fixed!
